File: src/main/java/com/qtech/bubbles/common/Percentage.java
package com.qtech.bubbles.common;

import org.jetbrains.annotations.NotNull;

import java.io.Serializable;

public record Percentage(double percentage) implements Serializable, Comparable<Percentage> {
    public static Percentage toPercentage(double value) {
        return new Percentage(value * 100);
    }

    public double getPercentage() {
        return percentage;
    }

    public double getValue() {
        return percentage / 100;
    }

    @Override
    public int compareTo(@NotNull Percentage o) {
        return Double.compare(percentage, o.percentage);
    }
}

File: src/main/kotlin/com/qtech/bubbles/common/Percentage.kt
package com.qtech.bubbles.common

import java.io.Serializable

class Percentage : Serializable, Comparable<Percentage> {
    val percentage: Double

    constructor(percentage: Double) {
        this.percentage = percentage * 100
    }

    constructor(value: Float) {
        percentage = value.toDouble()
    }

    val value: Double
        get() = percentage / 100

    override fun compareTo(other: Percentage): Int {
        return java.lang.Double.compare(percentage, other.percentage)
    }
}